service/tier1.go

```diff
@@ -568,9 +570,26 @@ func (s *Tier1Service) blocks(ctx context.Context, request *pbsubstreamsrpc.Requ
 		streamHandler = pipe
 	}
 
+	// We need to meter the bytes read from the live stream, as they are not metered by the dstore meters
+	blockMeteredHandlerFunc := bstream.HandlerFunc(func(blk *pbbstream.Block, obj interface{}) error {
+		step := obj.(bstream.Stepable).Step()
+		switch step {
+		case bstream.StepNewIrreversible:
+			// already metered by the store in compressed bytes.
+			// we will meter them here for potential future use
+			dmetering.GetBytesMeter(ctx).CountInc("uncompressed_bytes_read", len(blk.Payload.GetValue()))
+		default:
+			// all other cases are live blocks to be metered
+			dmetering.GetBytesMeter(ctx).AddBytesRead(len(blk.Payload.GetValue()))
+			dmetering.GetBytesMeter(ctx).CountInc("uncompressed_live_bytes_read", len(blk.Payload.GetValue()))
+		}
+
+		return streamHandler.ProcessBlock(blk, obj)
+	})
+
```

Comment:
ok, donc le tier2 va lire des blocks, et ici c'est la source elle-même qui va être metered, avec des métriques différentes, fack on fera l'aggrégation à la sortie, correct?

Reply:
the tier2 doesn't play into this.  tier2 does not have live blocks and so the blockstream doesn't need to be instrumented.  all of the tier2's data sources are dstores which are metered.